src/rules/sort-keys.js

```diff
@@ -41,36 +39,60 @@ const hasNonWhitespace = /\S/u;
 const comparators = {
```

Comment:
I think to circumvent, which are in my opinion, useless comments, we could add the type to the variable instead of each function with:
`Record<'ascending' | 'descending', Record<'alphanumeric' | 'natural', Record<'sensitive' | 'insensitive', Comparator>>>`. The type of the keys (e.g. `'alphanumeric' | 'natural'` could be their own type and be used for the converted option variables (e.g. `sortName` on line 166)